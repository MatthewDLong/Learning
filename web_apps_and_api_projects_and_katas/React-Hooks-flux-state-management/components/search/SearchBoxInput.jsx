import React, { useContext } from "react";

import {
  clearSearchResults,
  fetchSearchResults,
  updateSearchTerm
} from "../../state/actions/search";

import { SearchContext } from "../../state/context/search";
import { searchbox__input } from "./search.css";

const SearchBoxInput = ({ maxResults }) => {
  const { searchTerm, dispatch } = useContext(SearchContext);

  const handleInput = event => {
    event.preventDefault();
    clearSearchResults(dispatch);
    const searchTerm = event.target.value;
    updateSearchTerm(searchTerm, dispatch);

    searchTerm.length >= 2 &&
      fetchSearchResults(searchTerm, maxResults, dispatch);
  };

  return (
    <input
      id="searchbox__input"
      className={searchbox__input}
      type="text"
      value={searchTerm}
      onChange={handleInput}
      name="name"
      placeholder="city, airport, station, region and district..."
    />
  );
};

export default SearchBoxInput;
