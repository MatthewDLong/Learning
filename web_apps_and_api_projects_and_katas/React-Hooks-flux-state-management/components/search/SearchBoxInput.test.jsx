import React from "react";
import SearchBoxInput from "./SearchBoxInput";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<SearchBoxInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
