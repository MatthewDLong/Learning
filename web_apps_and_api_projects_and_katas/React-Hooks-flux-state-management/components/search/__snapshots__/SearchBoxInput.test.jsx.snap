// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="searchbox__input"
  id="searchbox__input"
  name="name"
  onChange={[Function]}
  placeholder="city, airport, station, region and district..."
  type="text"
/>
`;
